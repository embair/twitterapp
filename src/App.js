import 'whatwg-fetch';
import React, { Component } from 'react';
import SearchBox from './SearchBox.js';
import { Table } from 'react-bootstrap';
//import Reactable from 'reactable';
//const Table = Reactable.Table;

class App extends Component {
    constructor() {
        super();
        this.state = {
            tweets : [],
            isLoading : false,
        }
        //Temporary
        this.onClick('reactjs');
    }

    onClick(searchUser) {
        //TODO url encode
        fetch("/tweets?u="+searchUser).then((response) => {
            return response.text();
        }).then((body)=> {
            this.setState({tweets:Array.from(JSON.parse(body))});
        });
    }

    render() {
        return (
            <div className="container">
                <SearchBox onClick={this.onClick.bind(this)}/>
                <Table striped bordered condensed hover>
                    <thead>
                         <tr><th>#</th><th>Text</th><th>Favs</th></tr>
                    </thead>
                    <tbody>
                         {this.state.tweets.map((tweet, i)=> {
                            return <tr key={i}><td>{i}</td><td>{tweet.text}</td><td>{tweet.favorite_count}</td></tr>;
                         })}
                    </tbody>
                </Table>
            </div>
        );
    }
}

export default App;
