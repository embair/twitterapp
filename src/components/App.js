import React, { Component } from 'react';
import { Alert, Modal, Panel, Form } from 'react-bootstrap';
import 'whatwg-fetch';

import Immutable from 'immutable';
import SearchBox from './SearchBox.js';
import filters from '../filters.js';
import FilterEditor from './FilterEditor.js';
import Stats from './Stats.js';
import TweetsTable from './TweetsTable.js';
import Tweet from '../Tweet.js';

// Root application component
class App extends Component {

    static propTypes = {
        // the URL (or subpath) from which the app will retrieve tweets via a simple GET request
        tweetsServiceURL: React.PropTypes.string
    }

    static defaultProps = {
        tweetsServiceURL: '/tweets'
    }

    constructor() {
        super();
        this.state = {
            // Immutable.List<Tweet> : list of all retrieved tweets (unsorted, unfiltered); see Tweet.js for tweet properties
            tweets : null,
            // whether the client is currently waiting for results to be fetched from server
            fetching : false,
            // active filter values (key is index of filter in this.filters)
            filterValues : Immutable.Map(),
            // whether the statistics modal window is currently displayed
            showStats : false,
            // currently displayed error message
            error: null,
        }
        // listing of available filters
        this.filters = [
            filters.date,
            filters.fullText,
            filters.tweetLength,
            filters.mentions,
            filters.hashtags,
            filters.favourites,
            filters.hashtag,
            filters.mention
        ];
    }

    fetchTweets(user) {
        this.setState({fetching:true, error:null});
        fetch(this.props.tweetsServiceURL+"?u="+encodeURIComponent(user)).then((response) => {
            if (!this.state.fetching) return; // we no longer want the data
            this.setState({fetching:false});
            if (response.status >= 200 && response.status < 300) {
                return response.text();
            } else if (response.status >= 400 && response.status < 500) {
                this.showError('No data for this user!');
            } else {
                this.showError('Ooops! Something went wrong while fetching user data from Twitter. (Code '+response.status+')');
            }
        }).then((body)=> {
            if (body) {
                this.loadTweets(Array.from(JSON.parse(body)));
            }
        });
    }

    // display error message in the results area
    showError(err) {
        this.setState({error: err});
    }

    // show statistics modal
    showStats() {
        this.setState({showStats : true});
    }

    // hide statistics modal
    hideStats() {
        this.setState({showStats : false});
    }

    // load tweets from the timeline object generated by Twitter API
    loadTweets(timeline) {
        var transformed = timeline.map((tweetData,i) => {
            return new Tweet(i,tweetData);
        });

        this.setState({tweets: Immutable.List(transformed)});
    }

    // forget results
    resetResults() {
        this.setState({tweets: null, error: null })
    }

    render() {

        const renderResults=() => {
            if (this.state.fetching) {
                // currently loading data - show nothing
                return;
            } else if (this.state.error) {
                // error occured while getting data - display it 
                return (
                    <Alert bsStyle="danger">
                      <p>{this.state.error}</p>
                    </Alert>
                );
            } else if (!this.state.tweets) {
                // waiting for user selection - show nothing
                return;
            } else {
                // display results
                // apply filters the tweet list
                var displayList = this.state.tweets.filter((item) => {
                        // ok only if all filters match
                        return this.filters.every((filter, i) => {
                            if (!filter.apply(item, this.state.filterValues.get(i))) {
                                return false;
                            }
                            return true;
                        });
                    })
                

                return (
                    <div>
                    <Panel
                        header="Filters"
                        collapsible
                    >
                        <Form horizontal bsSize="small">
                        {this.filters.map((filter, key) => { return (
                            <FilterEditor key={key}
                                label={filter.label}
                                value={this.state.filterValues.get(key)}
                                placeholder={filter.prompt}
                                onChange={(value) => {
                                    this.setState({filterValues : this.state.filterValues.set(key, value)});
                                }}
                            />
                        )})}
                        </Form>
                    </Panel>
                    <a href="#" onClick={this.showStats.bind(this)}>Show statistics</a>
                    <TweetsTable tweets={displayList}/>
                    <Modal show={this.state.showStats} onHide={this.hideStats.bind(this)}>
                        <Modal.Header closeButton>
                            <Modal.Title>Statistics</Modal.Title>
                        </Modal.Header>
                        <Modal.Body>
                            <Stats tweets={this.state.tweets}/>
                        </Modal.Body>
                    </Modal> 
                    </div>               
                );            
            }

        }

        return (
            <div className="container">
                <h2>Twitter browser demo</h2>
                <SearchBox 
                    fetching={this.state.fetching}
                    onSubmit={this.fetchTweets.bind(this)}
                    onChange={this.resetResults.bind(this)}
                />
                {renderResults()}
            </div>
        );
    }
}

export default App;
