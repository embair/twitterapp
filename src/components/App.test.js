import React from 'react';
import App from './App';
import moment from 'moment';
import { mount } from 'enzyme';
import Immutable from 'immutable';

function createMockTweets(...array) {
    return Immutable.List(array.map( (item, i)=>{
        var text = item[0];
        var mentions = text.match(/@[^@ ]+/g) || [];
        var hashtags = text.match(/#[^# ]+/g) || [];
        return {
            id: i,
            text: item[0],
            favs: item[1],
            date: moment('2016-01-01T'+item[2]),
            dateString: '1.1.2016 '+item[2],
            mentions: mentions.map(str => str.slice(1)),
            hashtags: hashtags.map(str => str.slice(1)),
        }
    }));
};

describe('<App />', () => {
    
    const mockTweets = createMockTweets(
        ['First tweet of the day',0,'8:00'],
        ['Tweet with one @mention',0,'9:00'],
        ['Tweet with a lot of likes #popular',112,'9:30'],
        ['@thatOneGuy @thatOtherGuy @thatOneToo whats up',0,'10:00'],
        ['The edgiest of tweets #sobrave #verytrendy #needsmorehashtags ',0,'10:00'],
    );

    it('renders consistently', () => {

        const wrapper = mount(<App />);

        // initial snapshot
        expect(wrapper.debug()).toMatchSnapshot();
        wrapper.setState({
            tweets: mockTweets,
        });

        // snapshot with tweets
        expect(wrapper.debug()).toMatchSnapshot();

        // table should have header + 5 rows
        expect(wrapper.find('tr').length).toBe(6);
        
        // test that first table row has expected content
        expect(wrapper.find('tr').at(1).find('td').at(1).text()).toBe('First tweet of the day');

    });
});
