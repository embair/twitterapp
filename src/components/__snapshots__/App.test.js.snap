exports[`<App /> renders consistently 1`] = `
"<App tweetsServiceURL=\"/tweets\">
  <div className=\"container\">
    <h2>
      Twitter browser demo
    </h2>
    <SearchBox fetching={false} onSubmit={[Function]} onChange={[Function]} placeholder=\"Search\">
      <form onSubmit={[Function]}>
        <FormGroup bsClass=\"form-group\">
          <div className=\"form-group\">
            <InputGroup bsClass=\"input-group\">
              <span className=\"input-group\">
                <FormControl type=\"text\" placeholder=\"Search\" onChange={[Function]} value=\"\" componentClass=\"input\" bsClass=\"form-control\">
                  <input placeholder=\"Search\" onChange={[Function]} value=\"\" type=\"text\" id={[undefined]} className=\"form-control\" />
                </FormControl>
                <InputGroupButton bsClass=\"input-group-btn\">
                  <span className=\"input-group-btn\">
                    <Button onClick={[Function]} bsStyle=\"primary\" disabled={false} active={false} block={false} bsClass=\"btn\">
                      <button onClick={[Function]} disabled={false} type=\"button\" className=\"btn btn-primary\">
                        <Glyphicon glyph=\"search\" bsClass=\"glyphicon\">
                          <span className=\"glyphicon glyphicon-search\" />
                        </Glyphicon>
                      </button>
                    </Button>
                  </span>
                </InputGroupButton>
              </span>
            </InputGroup>
          </div>
        </FormGroup>
      </form>
    </SearchBox>
  </div>
</App>"
`;

exports[`<App /> renders consistently 2`] = `
"<App tweetsServiceURL=\"/tweets\">
  <div className=\"container\">
    <h2>
      Twitter browser demo
    </h2>
    <SearchBox fetching={false} onSubmit={[Function]} onChange={[Function]} placeholder=\"Search\">
      <form onSubmit={[Function]}>
        <FormGroup bsClass=\"form-group\">
          <div className=\"form-group\">
            <InputGroup bsClass=\"input-group\">
              <span className=\"input-group\">
                <FormControl type=\"text\" placeholder=\"Search\" onChange={[Function]} value=\"\" componentClass=\"input\" bsClass=\"form-control\">
                  <input placeholder=\"Search\" onChange={[Function]} value=\"\" type=\"text\" id={[undefined]} className=\"form-control\" />
                </FormControl>
                <InputGroupButton bsClass=\"input-group-btn\">
                  <span className=\"input-group-btn\">
                    <Button onClick={[Function]} bsStyle=\"primary\" disabled={false} active={false} block={false} bsClass=\"btn\">
                      <button onClick={[Function]} disabled={false} type=\"button\" className=\"btn btn-primary\">
                        <Glyphicon glyph=\"search\" bsClass=\"glyphicon\">
                          <span className=\"glyphicon glyphicon-search\" />
                        </Glyphicon>
                      </button>
                    </Button>
                  </span>
                </InputGroupButton>
              </span>
            </InputGroup>
          </div>
        </FormGroup>
      </form>
    </SearchBox>
    <div>
      <Panel header=\"Filters\" collapsible={true} defaultExpanded={false} bsStyle=\"default\" bsClass=\"panel\">
        <div className=\"panel panel-default\" id={{...}}>
          <div className=\"panel-heading\">
            <h4 role=\"presentation\" className=\"panel-title\">
              <a role={[undefined]} href={[undefined]} onClick={[Function]} aria-controls={[undefined]} aria-expanded={false} aria-selected={false}>
                Filters
              </a>
            </h4>
          </div>
          <Collapse in={false} onEnter={[undefined]} onEntering={[undefined]} onEntered={[undefined]} onExit={[undefined]} onExiting={[undefined]} onExited={[undefined]} timeout={300} unmountOnExit={false} transitionAppear={false} dimension=\"height\" getDimensionValue={[Function]}>
            <Transition in={false} onExited={[Function]} timeout={300} unmountOnExit={false} transitionAppear={false} aria-expanded={{...}} className=\"\" exitedClassName=\"collapse\" exitingClassName=\"collapsing\" enteredClassName=\"collapse in\" enteringClassName=\"collapsing\" onEnter={[Function]} onEntering={[Function]} onEntered={[Function]} onExit={[Function]} onExiting={[Function]}>
              <div id={[undefined]} role={[undefined]} className=\"panel-collapse collapse\" aria-hidden={true} aria-expanded={{...}}>
                <div className=\"panel-body\">
                  <Form horizontal={true} bsSize=\"small\" inline={false} componentClass=\"form\" bsClass=\"form\">
                    <form className=\"form-horizontal\">
                      <FilterEditor label=\"Sent date\" value={[undefined]} placeholder=\"Enter date\" onChange={[Function]}>
                        <FormGroup bsSize=\"small\" bsClass=\"form-group\">
                          <div className=\"form-group form-group-sm\">
                            <Col sm={3} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-3\">
                                <ControlLabel srOnly={false} bsClass=\"control-label\">
                                  <label htmlFor={[undefined]} className=\"control-label\">
                                    Sent date
                                  </label>
                                </ControlLabel>
                              </div>
                            </Col>
                            <Col sm={9} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-9\">
                                <FormControl type=\"text\" placeholder=\"Enter date\" onChange={[Function]} componentClass=\"input\" bsClass=\"form-control\">
                                  <input placeholder=\"Enter date\" onChange={[Function]} type=\"text\" id={[undefined]} className=\"form-control\" />
                                </FormControl>
                              </div>
                            </Col>
                          </div>
                        </FormGroup>
                      </FilterEditor>
                      <FilterEditor label=\"Contains\" value={[undefined]} placeholder=\"Enter text\" onChange={[Function]}>
                        <FormGroup bsSize=\"small\" bsClass=\"form-group\">
                          <div className=\"form-group form-group-sm\">
                            <Col sm={3} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-3\">
                                <ControlLabel srOnly={false} bsClass=\"control-label\">
                                  <label htmlFor={[undefined]} className=\"control-label\">
                                    Contains
                                  </label>
                                </ControlLabel>
                              </div>
                            </Col>
                            <Col sm={9} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-9\">
                                <FormControl type=\"text\" placeholder=\"Enter text\" onChange={[Function]} componentClass=\"input\" bsClass=\"form-control\">
                                  <input placeholder=\"Enter text\" onChange={[Function]} type=\"text\" id={[undefined]} className=\"form-control\" />
                                </FormControl>
                              </div>
                            </Col>
                          </div>
                        </FormGroup>
                      </FilterEditor>
                      <FilterEditor label=\"Length\" value={[undefined]} placeholder=\"Enter minimal length of tweets\" onChange={[Function]}>
                        <FormGroup bsSize=\"small\" bsClass=\"form-group\">
                          <div className=\"form-group form-group-sm\">
                            <Col sm={3} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-3\">
                                <ControlLabel srOnly={false} bsClass=\"control-label\">
                                  <label htmlFor={[undefined]} className=\"control-label\">
                                    Length
                                  </label>
                                </ControlLabel>
                              </div>
                            </Col>
                            <Col sm={9} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-9\">
                                <FormControl type=\"text\" placeholder=\"Enter minimal length of tweets\" onChange={[Function]} componentClass=\"input\" bsClass=\"form-control\">
                                  <input placeholder=\"Enter minimal length of tweets\" onChange={[Function]} type=\"text\" id={[undefined]} className=\"form-control\" />
                                </FormControl>
                              </div>
                            </Col>
                          </div>
                        </FormGroup>
                      </FilterEditor>
                      <FilterEditor label=\"Num. of @mentions\" value={[undefined]} placeholder=\"Enter minimal number of @mentions in tweet\" onChange={[Function]}>
                        <FormGroup bsSize=\"small\" bsClass=\"form-group\">
                          <div className=\"form-group form-group-sm\">
                            <Col sm={3} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-3\">
                                <ControlLabel srOnly={false} bsClass=\"control-label\">
                                  <label htmlFor={[undefined]} className=\"control-label\">
                                    Num. of @mentions
                                  </label>
                                </ControlLabel>
                              </div>
                            </Col>
                            <Col sm={9} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-9\">
                                <FormControl type=\"text\" placeholder=\"Enter minimal number of @mentions in tweet\" onChange={[Function]} componentClass=\"input\" bsClass=\"form-control\">
                                  <input placeholder=\"Enter minimal number of @mentions in tweet\" onChange={[Function]} type=\"text\" id={[undefined]} className=\"form-control\" />
                                </FormControl>
                              </div>
                            </Col>
                          </div>
                        </FormGroup>
                      </FilterEditor>
                      <FilterEditor label=\"Num. of #hastags\" value={[undefined]} placeholder=\"Enter minimal number of #hashtags in tweet\" onChange={[Function]}>
                        <FormGroup bsSize=\"small\" bsClass=\"form-group\">
                          <div className=\"form-group form-group-sm\">
                            <Col sm={3} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-3\">
                                <ControlLabel srOnly={false} bsClass=\"control-label\">
                                  <label htmlFor={[undefined]} className=\"control-label\">
                                    Num. of #hastags
                                  </label>
                                </ControlLabel>
                              </div>
                            </Col>
                            <Col sm={9} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-9\">
                                <FormControl type=\"text\" placeholder=\"Enter minimal number of #hashtags in tweet\" onChange={[Function]} componentClass=\"input\" bsClass=\"form-control\">
                                  <input placeholder=\"Enter minimal number of #hashtags in tweet\" onChange={[Function]} type=\"text\" id={[undefined]} className=\"form-control\" />
                                </FormControl>
                              </div>
                            </Col>
                          </div>
                        </FormGroup>
                      </FilterEditor>
                      <FilterEditor label=\"Num. of likes\" value={[undefined]} placeholder=\"Enter minimal number of likes\" onChange={[Function]}>
                        <FormGroup bsSize=\"small\" bsClass=\"form-group\">
                          <div className=\"form-group form-group-sm\">
                            <Col sm={3} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-3\">
                                <ControlLabel srOnly={false} bsClass=\"control-label\">
                                  <label htmlFor={[undefined]} className=\"control-label\">
                                    Num. of likes
                                  </label>
                                </ControlLabel>
                              </div>
                            </Col>
                            <Col sm={9} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-9\">
                                <FormControl type=\"text\" placeholder=\"Enter minimal number of likes\" onChange={[Function]} componentClass=\"input\" bsClass=\"form-control\">
                                  <input placeholder=\"Enter minimal number of likes\" onChange={[Function]} type=\"text\" id={[undefined]} className=\"form-control\" />
                                </FormControl>
                              </div>
                            </Col>
                          </div>
                        </FormGroup>
                      </FilterEditor>
                      <FilterEditor label=\"Contains #hashtag\" value={[undefined]} placeholder=\"Enter specific @hashtag\" onChange={[Function]}>
                        <FormGroup bsSize=\"small\" bsClass=\"form-group\">
                          <div className=\"form-group form-group-sm\">
                            <Col sm={3} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-3\">
                                <ControlLabel srOnly={false} bsClass=\"control-label\">
                                  <label htmlFor={[undefined]} className=\"control-label\">
                                    Contains #hashtag
                                  </label>
                                </ControlLabel>
                              </div>
                            </Col>
                            <Col sm={9} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-9\">
                                <FormControl type=\"text\" placeholder=\"Enter specific @hashtag\" onChange={[Function]} componentClass=\"input\" bsClass=\"form-control\">
                                  <input placeholder=\"Enter specific @hashtag\" onChange={[Function]} type=\"text\" id={[undefined]} className=\"form-control\" />
                                </FormControl>
                              </div>
                            </Col>
                          </div>
                        </FormGroup>
                      </FilterEditor>
                      <FilterEditor label=\"Contains @mention\" value={[undefined]} placeholder=\"Enter specific @mention\" onChange={[Function]}>
                        <FormGroup bsSize=\"small\" bsClass=\"form-group\">
                          <div className=\"form-group form-group-sm\">
                            <Col sm={3} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-3\">
                                <ControlLabel srOnly={false} bsClass=\"control-label\">
                                  <label htmlFor={[undefined]} className=\"control-label\">
                                    Contains @mention
                                  </label>
                                </ControlLabel>
                              </div>
                            </Col>
                            <Col sm={9} componentClass=\"div\" bsClass=\"col\">
                              <div className=\"col-sm-9\">
                                <FormControl type=\"text\" placeholder=\"Enter specific @mention\" onChange={[Function]} componentClass=\"input\" bsClass=\"form-control\">
                                  <input placeholder=\"Enter specific @mention\" onChange={[Function]} type=\"text\" id={[undefined]} className=\"form-control\" />
                                </FormControl>
                              </div>
                            </Col>
                          </div>
                        </FormGroup>
                      </FilterEditor>
                    </form>
                  </Form>
                </div>
              </div>
            </Transition>
          </Collapse>
        </div>
      </Panel>
      <a href=\"#\" onClick={[Function]}>
        Show statistics
      </a>
      <TweetsTable tweets={{...}}>
        <Table striped={true} bordered={true} condensed={true} hover={true} responsive={false} bsClass=\"table\">
          <table className=\"table table-striped table-bordered table-condensed table-hover\">
            <thead>
              <tr>
                <th className=\"clickable no-wrap\" onClick={[Function]}>
                  <Glyphicon glyph=\"chevron-up\" bsClass=\"glyphicon\">
                    <span className=\"glyphicon glyphicon-chevron-up\" />
                  </Glyphicon>
                  #
                </th>
                <th className=\"clickable no-wrap\" onClick={[Function]}>
                  text
                </th>
                <th className=\"clickable no-wrap\" onClick={[Function]}>
                  <Glyphicon glyph=\"star\" bsClass=\"glyphicon\">
                    <span className=\"glyphicon glyphicon-star\" />
                  </Glyphicon>
                </th>
                <th className=\"clickable no-wrap\" onClick={[Function]}>
                  sent
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>
                  0
                </td>
                <td>
                  First tweet of the day
                </td>
                <td>
                  0
                </td>
                <td>
                  1.1.2016 8:00
                </td>
              </tr>
              <tr>
                <td>
                  1
                </td>
                <td>
                  Tweet with one @mention
                </td>
                <td>
                  0
                </td>
                <td>
                  1.1.2016 9:00
                </td>
              </tr>
              <tr>
                <td>
                  2
                </td>
                <td>
                  Tweet with a lot of likes #popular
                </td>
                <td>
                  112
                </td>
                <td>
                  1.1.2016 9:30
                </td>
              </tr>
              <tr>
                <td>
                  3
                </td>
                <td>
                  @thatOneGuy @thatOtherGuy @thatOneToo whats up
                </td>
                <td>
                  0
                </td>
                <td>
                  1.1.2016 10:00
                </td>
              </tr>
              <tr>
                <td>
                  4
                </td>
                <td>
                  The edgiest of tweets #sobrave #verytrendy #needsmorehashtags 
                </td>
                <td>
                  0
                </td>
                <td>
                  1.1.2016 10:00
                </td>
              </tr>
            </tbody>
          </table>
        </Table>
      </TweetsTable>
      <Modal show={false} onHide={[Function]} backdrop={true} keyboard={true} autoFocus={true} enforceFocus={true} manager={{...}} renderBackdrop={[Function]} animation={true} dialogComponentClass={[Function]} bsClass=\"modal\">
        <Modal onHide={[Function]} keyboard={true} autoFocus={true} enforceFocus={true} manager={{...}} renderBackdrop={[Function]} show={false} onEntering={[Function]} onExited={[Function]} backdrop={true} backdropClassName=\"modal-backdrop\" containerClassName=\"modal-open\" transition={[Function]} dialogTransitionTimeout={300} backdropTransitionTimeout={150} />
      </Modal>
    </div>
  </div>
</App>"
`;
